### ISSUE TITLE
Types are wrong - break with TS v4

### ISSUE THREAD

**Issue by enspandi** [buschtoens/ember-link#742]
When running a project with typescript v4 I'm getting these errors:

```
... is defined as an accessor in class 'Link', but is overridden here in 'TestLink' as an instance property.
```

<details>

<summary>Click here</summary>

```
../../node_modules/ember-link/test-support/test-link.d.ts:4:5 - error TS2610: 'isActive' is defined as an accessor in class 'Link', but is overridden here in 'TestLink' as an instance property.

4     isActive: boolean;
      ~~~~~~~~

../../node_modules/ember-link/test-support/test-link.d.ts:5:5 - error TS2610: 'isActiveWithoutQueryParams' is defined as an accessor in class 'Link', but is overridden here in 'TestLink' as an instance property.

5     isActiveWithoutQueryParams: boolean;
      ~~~~~~~~~~~~~~~~~~~~~~~~~~

../../node_modules/ember-link/test-support/test-link.d.ts:6:5 - error TS2610: 'isActiveWithoutModels' is defined as an accessor in class 'Link', but is overridden here in 'TestLink' as an instance property.

6     isActiveWithoutModels: boolean;
      ~~~~~~~~~~~~~~~~~~~~~

../../node_modules/ember-link/test-support/test-link.d.ts:7:5 - error TS2610: 'isEntering' is defined as an accessor in class 'Link', but is overridden here in 'TestLink' as an instance property.

7     isEntering: boolean;
      ~~~~~~~~~~

../../node_modules/ember-link/test-support/test-link.d.ts:8:5 - error TS2610: 'isExiting' is defined as an accessor in class 'Link', but is overridden here in 'TestLink' as an instance property.

8     isExiting: boolean;
      ~~~~~~~~~

../../node_modules/ember-link/test-support/test-link.d.ts:9:5 - error TS2610: 'url' is defined as an accessor in class 'Link', but is overridden here in 'TestLink' as an instance property.

9     url: string;
      ~~~
```

</details>

---

Checking https://www.npmjs.com/package/ember-link?activeTab=explore the generated typescript definition file is not compatible with TS v4: `test-support/test-support/test-link.d.ts`

It should actually be like this:

```ts
import Transition from '@ember/routing/-private/transition';
import { Link } from 'ember-link';

export default class TestLink extends Link {
    get isActive(): boolean;
    get isActiveWithoutQueryParams(): boolean;
    get isActiveWithoutModels(): boolean;
    get isEntering(): boolean;
    get isExiting(): boolean;
    get url(): string;
    onTransitionTo?(): void;
    onReplaceWith?(): void;
    get qualifiedRouteName(): string;
    transitionTo(event?: Event): Transition;
    replaceWith(event?: Event): Transition;
    private _preventTransitionOut;
    private _createDummyTransition;
}
```

Can we pls either patch the `test-link.d.ts` and publish new version or upgrade typescript to v4?

**Comment by gossi** [comment]
The current solution is to "patch" it locally:

```diff
diff --git a/node_modules/ember-link/test-support/test-link.d.ts b/node_modules/ember-link/test-support/test-link.d.ts
index 4f6bcca..1c0421e 100644
--- a/node_modules/ember-link/test-support/test-link.d.ts
+++ b/node_modules/ember-link/test-support/test-link.d.ts
@@ -1,11 +1,17 @@
 import Transition from '@ember/routing/-private/transition';
 import { Link } from 'ember-link';
 export default class TestLink extends Link {
+    // @ts-expect-error
     isActive: boolean;
+    // @ts-expect-error
     isActiveWithoutQueryParams: boolean;
+    // @ts-expect-error
     isActiveWithoutModels: boolean;
+    // @ts-expect-error
     isEntering: boolean;
+    // @ts-expect-error
     isExiting: boolean;
+    // @ts-expect-error
     url: string;
     onTransitionTo?(): void;
     onReplaceWith?(): void;
```

**Comment by gossi** [comment]
I gonna close, as this is considered fixed with release of `v2.1.0`
